contracts/cw-core/src/msg.rs

```diff
@@ -100,9 +100,26 @@ pub enum ExecuteMsg {
     /// item already exists the existing value is overriden. If the
     /// item does not exist a new item is added.
     SetItem { key: String, addr: String },
-    /// Callable by the Admin, if one is configured, changes the
-    /// current admin. Setting to None removes the Admin.
-    UpdateAdmin { admin: Option<Addr> },
+    /// Callable by the admin of the contract. If ADMIN is None
+    /// removes the admin from the contract. If ADMIN is Some a new
+    /// admin is proposed and that new admin may become the admin by
+    /// executing the `AcceptAdminNomination` message.
+    ///
+    /// If there is already a pending admin nomination the
+    /// `WithdrawAdminNomination` message must be executed before a
+    /// new admin may be nominated.
+    NominateAdmin { admin: Option<String> },
```

Comment:
Just wondering why we are switching from an `Addr` to a `String` for the admin address?

Reply:
Great question! CosmWasm doesn't actually validate addresses while deserializing them. The result is (before) something like `UpdateAdmin { admin: "foo" }` would actually be a valid message to the contract. This is sub-optimal so we parse into a string so that the type system will make us validate the addresses before we can use them as addresses.